addons/mail/models/mail_message.py

```diff
@@ -28,9 +28,20 @@ class Message(models.Model):
     _message_read_limit = 30
 
     @api.model
-    def _get_default_from(self):
+    def _get_default_from(self, values=None):
+        if values is None or values is False:
+            values = {}
+        if self.env.user.id == SUPERUSER_ID:
+            company = self.env['res.company'].browse(self._context.get('force_company') or self._context.get('company_id')) or self.env.user.company_id
+            if values.get('model') and values.get('res_id'):
+                model = self.env[values['model']].browse(values['res_id'])
+                company = (model.company_id or company) if hasattr(model, 'company_id') else company
+            format_name = '%s (%s)'
```

Comment:
I'm not fan of the format used
something like that?
 - `{name} - {company}`
 - `{name} @ {company}`
 - `{name}, {company}`

Reply:
Why not, I was not especially inspired for the format either ;)
I guess this is a minor "implementation detail" :D